moto/sagemaker/models.py

```diff
@@ -2665,6 +2746,293 @@ def update_endpoint_weights_and_capacities(
+        model_package_description = model_package.model_package_description
+        creation_time = model_package.creation_time
+        inference_specification = model_package.inference_specification
+        source_algorithm_specification = model_package.source_algorithm_specification
+        validation_specification = model_package.validation_specification
+        model_package_status = model_package.model_package_status
+        model_package_status_details = model_package.model_package_status_details
+        certify_for_marketplace = model_package.certify_for_marketplace
+        model_approval_status = model_package.model_approval_status
+        created_by = model_package.created_by
+        metadata_properties = model_package.metadata_properties
+        model_metrics = model_package.model_metrics
+        last_modified_time = model_package.last_modified_time
+        last_modified_by = model_package.last_modified_by
+        approval_description = model_package.approval_description
+        customer_metadata_properties = model_package.customer_metadata_properties
+        drift_check_baselines = model_package.drift_check_baselines
+        domain = model_package.domain
+        task = model_package.task
+        sample_payload_url = model_package.sample_payload_url
+        additional_inference_specifications = (
+            model_package.additional_inference_specifications
+        )
+        return (
+            model_package_name,
+            model_package_group_name,
+            model_package_version,
+            model_package_arn,
+            model_package_description,
+            creation_time,
+            inference_specification,
+            source_algorithm_specification,
+            validation_specification,
+            model_package_status,
+            model_package_status_details,
+            certify_for_marketplace,
+            model_approval_status,
+            created_by,
+            metadata_properties,
+            model_metrics,
+            last_modified_time,
+            last_modified_by,
+            approval_description,
+            customer_metadata_properties,
+            drift_check_baselines,
+            domain,
+            task,
+            sample_payload_url,
+            additional_inference_specifications,
+        )
+
+    def create_model_package(
+        self,
+        model_package_name,
+        model_package_group_name,
+        model_package_description,
+        inference_specification,
+        validation_specification,
+        source_algorithm_specification,
+        certify_for_marketplace,
+        tags,
+        model_approval_status,
+        metadata_properties,
+        model_metrics,
+        client_token,
+        customer_metadata_properties,
+        drift_check_baselines,
+        domain,
+        task,
+        sample_payload_url,
+        additional_inference_specifications,
+    ):
+        # implement here
+        model_package_arn = arn_formatter(
```

Comment:
https://github.com/getmoto/moto/pull/6454/commits/d03894aad33a2c2f74f0bb74e64076618dcd384f

Reply:
I didnt move the model_package_version logic to the class because I need the backend to calculate the model_package_version itself. Moving this logic to the class would mean copying the list of modelpackages in each class which I think is not very sexy

Reply:
Makes sense